applications/calculator/main.cpp

```diff
@@ -25,148 +25,191 @@ enum CALCULATOR_OPERATOR
     OPERATOR_CLEAR_ALL = 6
 };
 
-void update_calculator_text()
+#define CALC_ID_BUTTON_0 "0"
+#define CALC_ID_BUTTON_1 "1"
+#define CALC_ID_BUTTON_2 "2"
+#define CALC_ID_BUTTON_3 "3"
+#define CALC_ID_BUTTON_4 "4"
+#define CALC_ID_BUTTON_5 "5"
+#define CALC_ID_BUTTON_6 "6"
+#define CALC_ID_BUTTON_7 "7"
+#define CALC_ID_BUTTON_8 "8"
+#define CALC_ID_BUTTON_9 "9"
+#define CALC_ID_BUTTON_PLUS "plus"
+#define CALC_ID_BUTTON_MINUS "minus"
+#define CALC_ID_BUTTON_EQUAL "equal"
+#define CALC_ID_BUTTON_CLEAR "clear"
+#define CALC_TEXT_BOX_NAME "calcscreen"
+
+class calculator_system
 {
-    if (selectionned_number == 1)
-    {
-        format_double(FORMAT_DECIMAL, main_number, (char *)calculator_text, 32);
-    }
-    else
+
+    Window *target;
+    double main_number = 0;
+    double second_number = 0;
+    int selectionned_number = 0;
+    int current_number_count = 0;
+    double current_operator = 0;
+    bool do_reset_after = true;
+    int calculator_text_length = 0;
+    char calculator_text[32];
+
+public:
+    void init_button()
     {
-        format_double(FORMAT_DECIMAL, second_number, (char *)calculator_text, 32);
+
+        CALCULATOR_BUTTON(CALC_ID_BUTTON_0, true, 0)
```

Comment:
It's not the Calculator object responsibility to managed windows events. Please respect the separation of concerns.